src/Interpreters/GetAggregatesVisitor.h

```diff
@@ -72,6 +73,14 @@ class GetAggregatesMatcher
 
             data.uniq_names.insert(column_name);
             data.aggregates.push_back(&node);
+            if ((node.name == "sum" || node.name == "avg" || node.name == "count") && node.arguments->children.size() == 1)
```

Comment:
This is probably going to work for window functions as well. By the way, do you need the aggregates visitor for this? For your purpose it should be enough to only match function names. Then you can save the the function nodes you have to change: a hash map from node arguments to the struct with three arrays: all count/sum/avg functions with these arguments. And then you could iterate over this hash map and perform the replacement where necessary.

Reply:
Do you mean I neednot add another variable in `GetAggregatesVisitor.h struct Data:fuse_sum_avg_count` or in `TreeRewriter.cpp:CustomizeFuseAggregateFunctionsVisitor`?

As you said, it is enough to  get a hash_map for all sum/count/avg functions  from `GetAggregatesVisitor.h struct Data:aggregates` but this variable is const, so I can not replace the function when necessary, So I think another visitor `TreeRewriter.cpp:CustomizeFuseAggregateFunctionsVisitor` is needed, 
Is this OK?

Reply:
I think a separate visitor that collects the necessary data is OK -- it will be totally self-contained, and you will be able to make only one pass over the AST (now you have two). So, in the end we'll have just one visitor and one function that passes over the resulting hash map.

Reply:
ok, I will rewrite the code.